test/built-ins/Iterator/prototype/Symbol.toStringTag/weird-setter.js

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2023 Michael Ficarra. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-set-iteratorprototype-@@tostringtag
+description: weird setter
+info: |
+  The value of the [[Get]] attribute is a built-in function that requires no arguments. It performs the following steps when called:
+    1. Return *"Iterator"*.
+
+  The value of the [[Set]] attribute is a built-in function that takes an argument _v_. It performs the following steps when called:
+    1. Perform ? SetterThatIgnoresPrototypeProperties(%Iterator.prototype%, %Symbol.toStringTag%, _v_).
+    2. Return *undefined*.
+features: [iterator-helpers]
+---*/
+
+let IteratorPrototype = Object.getPrototypeOf(Object.getPrototypeOf([][Symbol.iterator]()))
+
+let sentinel = 'a';
+
+let { get, set } = Object.getOwnPropertyDescriptor(Iterator.prototype, Symbol.toStringTag);
+
+assert.sameValue(Iterator.prototype[Symbol.toStringTag], 'Iterator');
+assert.sameValue(get.call(), 'Iterator');
+
+// 1. If _this_ is not an Object, then
+//   1. Throw a *TypeError* exception.
+assert.throws(() => set.call(undefined, ''));
```

Comment:
This test and the corresponding one for `constructor` property are broken. The first argument of `assert.throws()` [should be](https://github.com/tc39/test262/blob/main/harness/assert.js#L71) the expected error (i.e. `TypeError` in this case).